test/target_bigquery_test.py

```diff
@@ -0,0 +1,206 @@
+                            },
+                            "sdc_recursive_object_array": {
+                                "type": ["null", "object", "array"],
+                                "items": {"$ref": "#/definitions/sdc_recursive_object_array"},
+                            },
+                        },
+                    },
+                    "key_properties": ["id"],
+                    "bookmark_properties": ["id"],
+                }
+            ),
+            json.dumps(
+                {
+                    "type": "STATE",
+                    "value": {
+                        "bookmarks": {
+                            "database-public-fruitimals": {
+                                "last_replication_method": "INCREMENTAL",
+                                "replication_key": "id",
+                                "version": 1573504566181,
+                            }
+                        },
+                        "currently_syncing": "database-public-fruitimals",
+                    },
+                }
+            ),
+        ]
+        + [
+            json.dumps(
+                {
+                    "type": "RECORD",
+                    "stream": "fruitimals",
+                    "record": {"decimal": Decimal(f"{x}.1"), "id": x, "name": "Pear"},
+                    "version": 1573504566181,
+                    "time_extracted": "2020-03-06T14:22:46.181933Z",
+                }
+            )
+            for x in range(11000)
+        ]
+        + [
+            json.dumps(
+                {
+                    "type": "STATE",
+                    "value": {
+                        "bookmarks": {
+                            "database-public-fruitimals": {
+                                "last_replication_method": "INCREMENTAL",
+                                "replication_key": "id",
+                                "version": 1573504566181,
+                                "replication_key_value": 12,
+                            }
+                        },
+                        "currently_syncing": None,
+                    },
+                }
+            )
+        ]
+    )
+
+    persist_lines_hybrid(project_id, dataset_id, lines=lines, validate_records=False)
+
+    assert check_bigquery(bigquery_client, table, lambda data: len(data) == 11000)
+
+
+def test_full_table(setup_bigquery_and_config, check_bigquery, do_sync):
+    project_id, bigquery_client, config_filename, dataset_id = setup_bigquery_and_config(
+        replication_method="FULL_TABLE"
+    )
+    table = f"{project_id}.{dataset_id}.fruitimals"
+
+    stdout = do_sync(f"{test_path}/tap-sample-full-table.json", config_filename)
+
+    assert 'version": 1573504566181' in stdout[0]
+    assert check_bigquery(bigquery_client, table, lambda data: len(data) == 6)
+
+
+def test_hybrid_with_full_table_reset(setup_bigquery_and_config, check_bigquery, do_sync):
```

Comment:
This test is failing currently due to the BigQuery bug: https://issuetracker.google.com/issues/152476581

But we might want to avoid using this method after all as we'd not have the row_ids set... I haven't committed the change to the ETL repo using it so I guess it's not necessarily a huge problem if this doesn't work and we could remove the test unless it gets fixed very soon.